RadixWallet/Clients/DeviceFactorSourceClient/DeviceFactorSourceClient+Interface.swift

```diff
@@ -42,23 +41,56 @@ struct DiscrepancyUnsupportedCurve: Swift.Error {}
 
 // MARK: - PublicKeysFromOnDeviceHDRequest
 public struct PublicKeysFromOnDeviceHDRequest: Sendable, Hashable {
-	public let deviceFactorSource: DeviceFactorSource
 	public let derivationPaths: [DerivationPath]
-	public let loadMnemonicPurpose: SecureStorageClient.LoadMnemonicPurpose
+
+	public func getMnemonicWithPassphrase() throws -> MnemonicWithPassphrase {
+		@Dependency(\.secureStorageClient) var secureStorageClient
+		switch source {
+		case let .privateHDFactorSource(privateHD):
+			return privateHD.mnemonicWithPassphrase
+		case let .loadMnemonicFor(deviceFactorSource, loadMnemonicPurpose):
+			let factorSourceID = deviceFactorSource.id
+			guard
+				let mnemonicWithPassphrase = try secureStorageClient
+				.loadMnemonic(factorSourceID: factorSourceID, purpose: loadMnemonicPurpose)
+			else {
+				loggerGlobal.critical("Failed to find factor source with ID: '\(factorSourceID)'")
+				throw FailedToFindFactorSource()
+			}
+			return mnemonicWithPassphrase
+		}
+	}
+
+	public enum Source: Sendable, Hashable {
+		case privateHDFactorSource(PrivateHDFactorSource)
```

Comment:
the important change is that it is now possible to pass in the mnemonic to `DeviceFactorSourcesClient`! via `PrivateHDFactorSource`.